core/src/main/java/hudson/slaves/Cloud.java

```diff
@@ -323,10 +390,8 @@ public HttpResponse doConfigSubmit(StaplerRequest req, StaplerResponse rsp) thro
             throw new ServletException("No such cloud " + this.name);
         }
         Cloud result = cloud.reconfigure(req, req.getSubmittedForm());
-        String proposedName = result.name;
-        if (!proposedName.equals(this.name)
-                && j.getCloud(proposedName) != null) {
-            throw new Descriptor.FormException(jenkins.agents.Messages.CloudSet_CloudAlreadyExists(proposedName), "name");
+        if (!(result.name).equals(this.name)) {
+            throw new Descriptor.FormException(jenkins.agents.Messages.Cloud_DoNotRename(), "name");
```

Comment:
Another option was to just ignore any name changes, i.e. `result.name  = this.name`

Reply:
Will this case get triggered by some existing cloud plugins until they override `isNameEditable` and update their form configurations to remove the related field?

Reply:
I guess that is the case described in https://github.com/jenkinsci/jenkins/pull/8310#issuecomment-1705680239. Any idea how widespread that kind of problem will be? Should we prepare downstream PRs?

Reply:
Yes, this exception will be thrown by existing cloud nodes from plugins that do not have the form updated. I'll take a look to see what plugins are affected.

Reply:
On an initial search of "extends Cloud " and "exnteds AbstractCloudImpl" , I see around 50+ plugins that need to be checked.

Reply:
Hmm, yeah I'm not sure how to pop a warning dialog here without having to update `config.jelly` in all the plugins.

For now I would be tempted to just ignore name changes in this context. It would be pretty frustrating to reconfigure various things and attempt to rename an object at the same time, hit save, get an exception, and then have to reconfigure things again. IDK though how common it is to rename and reconfigure things at the same time. If people normally do them separately then the current behavior at least gives you clear feedback on why the rename did not happen vs just silently ignoring the rename.

Reply:
(or file PRs into the major cloud plugins)

Reply:
I tried without any success to get the submit button to be blocked if the name is changed. Looks like I would have to use the `button` tag instead, but I think I was getting too lost in the weeds.

I think I'll go ahead and just block the name change without the exception. I don't think name changes happen that much as this bug has only been discovered now. I think the fact that the rename page exists should be enough of a clue. I'll also go ahead and try and file some downstream PRs based off of this incremental.